src/CloudNative.CloudEvents/CloudEventAttributeType.cs

```diff
@@ -86,15 +132,16 @@ public abstract class CloudEventAttributeType
         /// <param name="value">The value to validate. Must be non-null, and suitable for this attribute type.</param>
         public abstract void Validate(object value);
 
-        private CloudEventAttributeType(string name, Type clrType)
+        private CloudEventAttributeType(string name, Ordinal ordinal, Type clrType)
         {
             Name = name;
```

Comment:
Can we do ordinal.ToString() for the name?

Reply:
Unfortunately not - the names include `"URI-Reference"` which wouldn't be a valid enum value. We could potentially have an attribute on the enum value to specify the name... but I suspect just hard-coding the names is simpler.